_posts/en/newsletters/2019-11-06-newsletter.md

```diff
@@ -0,0 +1,152 @@
+
+    Gleb Naumenko sent an [email][naumenko addr relay] to the
+    Bitcoin-Dev mailing list suggesting that nodes and clients should
+    signal to their peers whether or not they want to participate in
+    address relay.  This will avoid wasting bandwidth on clients that
+    don't want the addresses and can make it easier to determine the
+    consequences of certain network behavior related to address relay.
+
+    Two methods are proposed for allowing nodes to indicate whether or
+    not they want to participate in address relay---a per-node method
+    and a per-connection method.  The per-node method could easily build
+    on top of work already being done for addrv2 messages (see [Newsletter
+    #37][news37 addrv2]), but it would be less flexible than the the
+    per-connection method.  In particular, the per-connection method
+    could allow a node to dedicate some connections to transaction relay
+    and other connections to address relay, producing possible privacy
+    advantages.  Naumenko's email requests feedback on which method would
+    be preferred by implementers of both full nodes and lightweight
+    clients.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #16943][] adds a `generatetodescriptor` RPC that allows
+  new blocks generated during testing (e.g. in regtest mode) to pay a
+  script represented by an [output script descriptor][].  Prior to this
+  change, only the `generatetoaddress` RPC was available and it can only
+  pay P2PKH, P2SH, or P2WSH addresses.
+
+- [C-Lightning #3220][] begins always creating signatures with a low *r*
+  value, reducing the maximum size of the signature by one byte and
+  saving about 0.125 vbytes on average per C-Lightning peer in onchain
+  transactions.  Bitcoin Core also previously adopted this change to its
+  wallet's signature-generation code (see [Newsletter #8][news8 lowr]).
+
+- [LND #3558][] synthesizes a unified policy for any case where two
+  particular nodes have multiple channels open between them and then
+  uses this synthetic policy when considering routing through any of
+  those channels.  [BOLT4][bolt4 non-strict rec] recommends that multiple channels
+  between the same nodes should all use the same policy, but this
+  doesn't always happen, so this change tries to determine "the greatest
+  common denominator of all policies" between the nodes.  Using a single
+  policy reduces the number of routes the node needs to evaluate when
+  making a payment.
+
+- [LND #3556][] adds a new `queryprob` RPC that returns the expected
+  probability that a payment would succeed given a particular
+  destination node and amount.  This replaces functionality previously
```

Comment:
IIUC the queryprob rpc takes source node, destination node, and amount as args.

Reply:
See https://github.com/joostjager/lnd/blob/fb57255b5c62a064c6f87565ed8c5ba805fd14a4/lnrpc/routerrpc/router.proto#L370